### PR TITLE
zenfs: Get zenfs files from zenfs CMakeLists.txt if possible

### PR DESCRIPTION
**Pull request by jsvaerke**
ZenFS may provide a CMakeLists.txt file describing the source
files required to build ZenFS. If present, use that instead
of parsing the plugin make file to get the source file names,
when building with ZenFS.

If no CMakeLists.txt file is provided by ZenFS, use existing
approach.

Signed-off-by: Jorgen Hansen <jorgen.hansen@wdc.com>

### CODE REVIEW

File: storage/rocksdb/CMakeLists.txt [hunk 1]
```diff
@@ -92,10 +92,17 @@ ELSE()
     SET(PLUGIN_ROOT "${ROCKSDB_PLUGINS_ROOT}/${PLUGIN}/")
     MESSAGE(STATUS "MyRocks: Building RocksDB plugin: ${PLUGIN}")
     FILE(GLOB PLUGINMKFILE "${PLUGIN_ROOT}${PLUGIN}*.mk")
+    FILE(GLOB PLUGINCMAKEFILE "${PLUGIN_ROOT}CMakeLists.txt")
     SET(CURRENT_PLUGIN_SOURCES)
     SET(CURRENT_PLUGIN_LD)
-    IF (NOT EXISTS ${PLUGINMKFILE})
-      MESSAGE(SEND_ERROR "Missing RocksDB plugin makefile in ${PLUGIN_ROOT}")
+    IF (EXISTS ${PLUGINCMAKEFILE})
+      ADD_SUBDIRECTORY("${PLUGIN_ROOT}")
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt:102]: @jsvaerke I tried to apply this patch to the upcoming release `8.0.30-22` branch and check it in both modes. The old approach with extracting file lists from the `zenfs.mk` works absolutely fine - I tested it with the `zenfs` submodule pointing to the `v2.1.0` release tag. However, when I advanced it the latest `master` (commit c81038def76eca58db380e7f45d98fb6a54d3ed2 "fs: make zenfs compile with c++11 again") I encountered a CMake configuration problem.
```
-- Checking for module 'libzbd>=1.5.0'
--   No package 'libzbd' found
CMake Error at /usr/share/cmake-3.22/Modules/FindPkgConfig.cmake:603 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPkgConfig.cmake:825 (_pkg_check_modules_internal)
  storage/rocksdb/rocksdb_plugins/zenfs/CMakeLists.txt:11 (pkg_check_modules)
```
The problem is with the
```
find_package(PkgConfig REQUIRED)
pkg_check_modules(ZBD REQUIRED libzbd>=1.5.0)
```
lines in the ZenFS's `CMakeLists.txt` coming from here
https://github.com/westerndigitalcorporation/zenfs/commit/7908084f811f4cd58fb798225ab119df289dadad#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR9-R10.
As Percona Server links `libzbd` statically from the submodule, we do not have this `pkg_check_modules(ZBD REQUIRED libzbd>=1.5.0)` dependency.
The fix would be to either completely remove those `pkg_check_modules()` lines or wrap them around certain CMake `IF(<NON_PERCONA_SERVER_BUILD>)` condition.
But unless this is done, I do not want to include this patch to the main Percona Server source tree.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: storage/rocksdb/CMakeLists.txt (hunk 1 above)
Comment (jsvaerke) [reply storage/rocksdb/CMakeLists.txt:102]: Thank you for the feedback. I'll look into addressing this on the ZenFS side.

**Comment by jsvaerke** [comment]
Sorry, my fork got messed up so it might be difficult to see the updates to the commit above  - zenfs master now allows disabling the linzbd check through setting ZENFS_SKIP_LIBZBD_CHECK to a non-zero value. So the patch for percona has been updated with the following:

>       IF (PLUGIN STREQUAL "zenfs")
>         SET(ZENFS_SKIP_LIBZBD_CHECK 1)
>       ENDIF()

...

>       UNSET(ZENFS_SKIP_LIBZBD_CHECK)

**Comment by percona-ysorokin** [comment]
Closing this PR in favour of https://github.com/percona/percona-server/pull/4901 (based on `release-8.0.30-22` branch).

**Comment by percona-ysorokin** [comment]
@jsvaerke 
The patch has been merged and will be present in the upcoming Percona Server 8.0.30-22 release.
Thanks for your contribution.

**Pull request closed without merging**